File: EduHubConnect-Backend-UsingSpringBoot/src/main/java/com/connect/eduHub/dto/UserProfileDTO.java
package com.connect.eduHub.dto;

import java.util.List;

import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserProfileDTO {
    private Long userId;
    private String userName;
    private String bio;
    private String collageName;
    private AddressDTO addressDTO;
    private String profileImageUrl;
    private String bannerImageUrl;
    private String about;

    private List<ExperienceDTO> experiences;
    private List<EducationDTO> educationList;
    private List<SkillDTO> skills;
   
    private List<PostDTO> selfPosts;
}

File: EduHubConnect-AndroidFrontEnd/app/src/main/java/com/connect/eduhubconnect/dto/UserProfileDTO.kt
package com.connect.eduhubconnect.dto

data class UserProfileDTO(
    val addressDTO: AddressDTO,
    val bannerImageUrl: String,
    val bio: String,
    val collageName: String,
    val educationList: List<EducationDTO>,
    val experiences: List<ExperienceDTO>,
    val profileImageUrl: String,
    val about:String,
    val selfPosts: List<SelfPostDTO>,
    val skills: List<SkillDTO>,
    val userId: Int,
    val userName: String
)